pkg/api/types.go

```diff
@@ -2143,6 +2143,20 @@ const (
 
 	// SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
 	SSHAuthPrivateKey = "ssh-privatekey"
+
+	// SecretTypeTLS contains information about SSL termination used with the Ingress resource.
+	//
+	// Required fields:
+	// - Secret.Data["key"] - TLS private key.
+	//   Secret.Data["cert"] - TLS certificate.
+	//   Secret.Data["chain"] - optional certificate chain.
+	// TODO: Consider supporting different formats, specifying CA/destinationCA.
+	SecretTypeTLS SecretType = "kubernetes.io/tls"
+
+	// TLSCertKey is the key for tls certificates in a TLS secert.
+	TLSCertKey = "cert"
```

Comment:
Your key-with-filename-extension idea makes sense, matches what loadbalancers expect and is similar to ".dockercfg". Modified pr to have tls.crt/key.